core/templates/dev/head/pages/exploration_player/PlaythroughServiceSpec.js

```diff
@@ -0,0 +1,252 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the playthrough service.
+ */
+
+describe('Playthrough service', function() {
+  beforeEach(module('oppia'));
+
+  describe('Test playthrough service functions', function() {
+    beforeEach(inject(function($injector) {
+      this.expId = 'expId1';
+      this.expVersion = 1;
+      this.ps = $injector.get('PlaythroughService');
+      this.laof = $injector.get('LearnerActionObjectFactory');
+      this.ps.initSession(this.expId, this.expVersion);
+    }));
+
+    it('should initialize a session with correct values.', function() {
+      var playthrough = this.ps.getPlaythrough();
+
+      expect(playthrough.expId, this.expId);
+      expect(playthrough.expVersion, this.expVersion);
+    });
+
+    it('should record exploration start action.', function() {
+      this.ps.recordExplorationStartAction('initStateName1');
+      var playthrough = this.ps.getPlaythrough();
+
+      expect(playthrough.actions).toEqual(
+        [this.laof.createFromBackendDict({
+          action_type: 'ExplorationStart',
+          action_customization_args: {
+            state_name: {
+              value: 'initStateName1'
+            }
+          },
+          schema_version: 1
+        })]);
+    });
+
+    it('should record answer submit action.', function() {
+      this.ps.recordAnswerSubmitAction(
+        'stateName1', 'stateName2', 'TextInput', 'Hello', 'Try again', '30');
+      var playthrough = this.ps.getPlaythrough();
+
+      expect(playthrough.actions).toEqual(
+        [this.laof.createFromBackendDict({
```

Comment:
Can you use `createNew` here now that you have it?

Reply:
Done.